src/rascal/structure_managers/property_typed.hh

```diff
@@ -319,7 +319,11 @@ namespace rascal {
      * Accessor for last pushed entry for dynamically sized properties
      */
     reference back() {
-      auto && index{this->values.size() - this->get_nb_comp()};
+      // -1 is correction for 0-start indexing
+      auto && index{this->values.size() / this->get_nb_comp() - 1};
+      if (index < 0) {
+        throw std::runtime_error("Property is empty, .back() is undefined.");
```

Comment:
Could you add a test that this error is thrown if the property is empty?

Reply:
You mean forcibly test for the throwing of the runtime error?

Reply:
yes, using `BOOST_CHECK_THROWS` https://www.boost.org/doc/libs/1_67_0/libs/test/doc/html/boost_test/utf_reference/testing_tool_ref/assertion_boost_level_throw.html

Reply:
Sure